src/Avalonia.Controls/MenuItem.cs

```diff
@@ -440,10 +446,17 @@ protected override void OnAttachedToLogicalTree(LogicalTreeAttachmentEventArgs e
         protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
         {
             base.OnAttachedToVisualTree(e);
-
+            SubscribeToParnetSizeScope();
```

Comment:
If you are subscribing in the OnAttachedToVisualTree, you can entirely skip `this.GetObservable(VisualParentProperty)+Switch` part and instead use something like:
```
_scopeBinding = this.Bind(DefinitionBase.PrivateSharedSizeScopeProperty, parent.GetObservable(DefinitionBase.PrivateSharedSizeScopeProperty))
```